js/util/pipeline.js

```diff
@@ -74,7 +102,48 @@ Pipeline.ErrorCode = {
   SEGMENT_VERIFICATION_FAILED: 5,
   NACK_RECEIVED: 6,
   NO_FINALBLOCK: 7,
-  MAX_NACK_TIMEOUT_RETRIES: 8
+  MAX_NACK_TIMEOUT_RETRIES: 8,
+  MISC: 9
+};
+
+/**
+ * Stop the pipeline
+ */
+Pipeline.prototype.cancel = function()
+{
+  if (this.isStopped)
+    return;
+
+  this.isStopped = true;
+};
+
+/**
+ * Make an Interest with proper name
+ * @description If baseInterest's name contains version number it will be used, otherwise the
+ *              discovered version number will be used. Then @param segNo is appended to the name.
+ *              If no version number is available, we send back a copy of baseInterest.
+ */
+Pipeline.prototype.makeInterest = function(segNo)
+{
+  var interest = new Interest(this.baseInterest);
```

Comment:
To use the Interest class in node.js, at the top of the file please add 

    var Interest = require('../interest.js').Interest;

Reply:
If you make this change, I think we're ready to merge.